modules/widgets/twitter-timeline.php

```diff
@@ -294,16 +312,11 @@ public function form( $instance ) {
 
 		$instance = wp_parse_args( (array) $instance, $defaults );
 
-		if ( empty( $instance['type'] ) ) {
-			// Decide the correct widget type.  If this is a pre-existing
-			// widget with a numeric widget ID, then the type should be
-			// 'widget-id', otherwise it should be 'profile'.
-			if ( ! empty( $instance['widget-id'] ) && is_numeric( $instance['widget-id'] ) ) {
-				$instance['type'] = 'widget-id';
-			} else {
-				$instance['type'] = 'profile';
-			}
+		if ( 'widget-id' === $instance['type'] ) {
+			$instance['widget-id'] = '';
 		}
+
+		$instance['type'] = 'profile';
```

Comment:
I'd move this up to L313 below the setting of `$instance`, but only for tidiness sake.

Reply:
Actually this needs to be set right after 
```php
if ( 'widget-id' === $instance['type'] ) {
	$instance['widget-id'] = '';
}
```

Reasoning is that first we want to empty the Widget ID value if the widget is of type Widget ID (in order to avoid having a widget of type Profile with a numeric, non-Twitter-username, ID), and only then force the widget to be of type Profile.